packages/legacy-api/controllers/Meditor.js

```diff
@@ -68,6 +69,9 @@ async function handlePublicationAcknowledgements(message) {
                 $pull: {
                     'x-meditor.publishedTo': {
                         target: acknowledgement.target,
+                        // if document is in "Published" state, this would clear out any publication statuses for **other** states
+                        // i.e. any publication statuses that are marked as "Draft" would be cleared out
+                        state: { $ne: acknowledgement.state },
```

Comment:
Only clear out publication statuses that don't match the acknowledgements state.

If an acknowledgement doesn't have a state, it would also be cleared out